htsget-config/src/storage/mod.rs

```diff
@@ -0,0 +1,199 @@
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+
+use crate::resolver::ResolveResponse;
+use crate::storage::local::LocalStorage;
+#[cfg(feature = "s3-storage")]
+use crate::storage::s3::S3Storage;
+use crate::types::{Query, Response, Result};
+
+pub mod local;
+#[cfg(feature = "s3-storage")]
+pub mod s3;
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+pub enum TaggedStorageTypes {
+  #[serde(alias = "local", alias = "LOCAL")]
+  Local,
+  #[cfg(feature = "s3-storage")]
+  #[serde(alias = "s3")]
+  S3,
+}
+
+/// If s3-storage is enabled, then the default is `S3`, otherwise it is `Local`.
+impl Default for TaggedStorageTypes {
+  #[cfg(not(feature = "s3-storage"))]
+  fn default() -> Self {
+    Self::Local
+  }
+
+  #[cfg(feature = "s3-storage")]
+  fn default() -> Self {
+    Self::S3
+  }
+}
+
```

Comment:
I'm not sure if it's the best idea to have this change in behaviour based on feature flags at compile time, but I think it is okay.

Reply:
Yeah, while reading the changes I think I'd go for no features by default, tbh, but then again, there should be some default storage backend... local seems a good simple candidate for default storage backend but our own default in production is S3 :-S

Reply:
Yeah, If you think this would cause confusion, I don't think it would be the worst to remove it, as it would only add one line for each `resolver` in the config.

Reply:
I think it should be fine as-is, let's change it if that confusion comes up?